setup.py

```diff
@@ -12,10 +12,11 @@
 
 setup(
     name='jediepcserver',
-    version='0.2.8',
+    version='0.3.0',
     py_modules=['jediepcserver'],
     install_requires=[
-        "jedi>=0.11.0",
+        "jedi>=0.11.0; python_version>'3.5'",
+        "jedi>=0.11.0,<0.18.0; python_version<='3.5'",
```

Comment:
I think it is supposed to be declared in Jedi as `python_requires`, and then pip dependency resolution will not return the newer version:
```
immerrr@mmrcomp:~$ python2 -m pip install --user 'jedi>=0.11' -U
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting jedi>=0.11
  Using cached https://files.pythonhosted.org/packages/c3/d4/36136b18daae06ad798966735f6c3fb96869c1be9f8245d2a8f556e40c36/jedi-0.17.2-py2.py3-none-any.whl
Collecting parso<0.8.0,>=0.7.0 (from jedi>=0.11)
  Downloading https://files.pythonhosted.org/packages/93/d1/e635bdde32890db5aeb2ffbde17e74f68986305a4466b0aa373b861e3f00/parso-0.7.1-py2.py3-none-any.whl (109kB)
    100% |████████████████████████████████| 112kB 4.0MB/s 
Installing collected packages: parso, jedi
  Found existing installation: parso 0.3.1
    Uninstalling parso-0.3.1:
      Successfully uninstalled parso-0.3.1
  Found existing installation: jedi 0.13.1
    Uninstalling jedi-0.13.1:
      Successfully uninstalled jedi-0.13.1
Successfully installed jedi-0.17.2 parso-0.7.1
```

Reply:
IIRC python_requires will declare the python version required by the package currently installed. python_version restrict the version for a dependency. So not doing exactly the same thing.

We can have the dependency jedi up to 0.17.2 for python2 and python3 (up to 3.5) then jedi from 0.11 for python 3.5+. So this is more for splitting the dependency version for python3 versions than python2: The latest jedi package available for python2 is 0.17.2 so the restriction used here isn't really applied in case of python2.

python3~3.5 users may have to adjust the jedi version between 0.11 and 0.17, or report here so we can make more precise restrictions : for example jedi changelog declare 3.4 support dropped in 0.17 but it may just be a 'we wont fix issues with 3.4 anymore' rather than 'code dont work' (python3 on debian is 3.7 so not sure if there is a lot of theses old python version used anyway).

Reply:
> IIRC python_requires will declare the python version required by the package currently installed. python_version restrict the version for a dependency. So not doing exactly the same thing.

Yes, what I mean is that jedi has a `python_requires` in its setup.py that will prevent pip from installing it on incompatible Pythons, so restricting python versions in downstream packages is not required.

Reply:
You're right, time for a coffee it seems. Don't know why I've first read your sentence as [diff on python_version] is supposed to be declared as python_requires ! Sorry ^^

Reply:
:+1: